00_homework_kawapa/PROJECT2/src/imager.h

```diff
@@ -0,0 +1,1769 @@
+    class ImageBuffer
+    {
+    public:
+        ImageBuffer (
+            size_t _pixelsWide, 
+            size_t _pixelsHigh, 
+            const Color &backgroundColor)
+                : pixelsWide(_pixelsWide)
+                , pixelsHigh(_pixelsHigh)
+                , numPixels(_pixelsWide * _pixelsHigh)
+        {
+            array = new PixelData[numPixels];
+        }
+
+        virtual ~ImageBuffer()
+        {
+            delete[] array;
+            array = NULL;
+            pixelsWide = pixelsHigh = numPixels = 0;
+        }
+
+        // Returns a read/write reference to the pixel data for the 
+        // specified column (i) and row (j).
+        // Throws an exception if the coordinates are out of bounds.
+        PixelData& Pixel(size_t i, size_t j) const
+        {
+            if ((i < pixelsWide) && (j < pixelsHigh))
+            {
+                return array[(j * pixelsWide) + i];
+            }
+            else
+            {
+                throw ImagerException("Pixel coordinate(s) out of bounds");
+            }
+        }
+
+        size_t GetPixelsWide() const
+        {
+            return pixelsWide;
+        }
+
+        size_t GetPixelsHigh() const
+        {
+            return pixelsHigh;
+        }
+
+        // Finds the maximum red, green, or blue value in the image.
+        // Used for automatically scaling the image brightness.
+        double MaxColorValue() const
+        {
+            double max = 0.0;
+            for (size_t i=0; i < numPixels; ++i) 
+            {
+                array[i].color.Validate();
+                if (array[i].color.red > max)
+                {
+                    max = array[i].color.red;
+                }
+                if (array[i].color.green > max)
+                {
+                    max = array[i].color.green;
+                }
+                if (array[i].color.blue > max)
+                {
+                    max = array[i].color.blue;
+                }
+            }
+            if (max == 0.0)
+            {
+                // Safety feature: the image is solid black anyway,
+                // so there is no point trying to scale it.
+                // If we did, we would end up dividing by zero.
+                max = 1.0;
+            }
+            return max;
+        }
+
+    private:        
+        size_t  pixelsWide;     // the width of the image in pixels (columns).
+        size_t  pixelsHigh;     // the height of the image in pixels (rows).
+        size_t  numPixels;      // the total number of pixels.
+        PixelData*  array;      // flattened array [pixelsWide * pixelsHigh].
+    };
+
+    // Output operators (print helpful debug information).
+    std::ostream& operator<< (std::ostream&, const Color&);
+    std::ostream& operator<< (std::ostream&, const Vector&);
+    std::ostream& operator<< (std::ostream&, const Intersection&);
+    void Indent(std::ostream&, int depth);
+}
+
+#endif // __DDC_IMAGER_H

```

Comment:
niestety imager.h to duży plik i nie wiem co tutaj zmieniłeś. Jakby w commicie było widać zmiany względem oryginalnej wersji to byłoby wygodniej.